rust/src/sip/parser.rs

```diff
@@ -65,7 +65,7 @@ pub enum Method {
 
 #[inline]
 fn is_token_char(b: u8) -> bool {
-    is_alphanumeric(b) || b"!%'*+-._`".contains(&b)
+    is_alphanumeric(b) || b"!%'*+-._`;=@".contains(&b)
```

Comment:
Why these new characters ?

Reply:
Because they are valid chars for the URI field, please look at the example below:
`REGISTER sip:192.168.43.100;transport=TCP SIP/2.0`

Reply:
Only for TCP ?
Is there a RFC referencing this ?

Reply:
No, TCP and UDP.
Yes, it's defined in SIP RFC page 222-223: https://www.rfc-editor.org/rfc/rfc3261

> 
> SIP-URI          =  "sip:" [ userinfo ] hostport
>                     uri-parameters [ headers ]
> userinfo         =  ( user / telephone-subscriber ) [ ":" password ] "@"
> user             =  1*( unreserved / escaped / user-unreserved )
> user-unreserved  =  "&" / "=" / "+" / "$" / "," / ";" / "?" / "/"
> password         =  *( unreserved / escaped /
>                     "&" / "=" / "+" / "$" / "," )
> hostport         =  host [ ":" port ]
> host             =  hostname / IPv4address / IPv6reference
> uri-parameters    =  *( ";" uri-parameter)
> uri-parameter     =  transport-param / user-param / method-param
>                      / ttl-param / maddr-param / lr-param / other-param
> transport-param   =  "transport="
>                      ( "udp" / "tcp" / "sctp" / "tls"
>                      / other-transport)
> ...
>

Reply:
No, because `:` is not a valid token character, and then it's included here:
```
#[inline]
fn is_request_uri_char(b: u8) -> bool {
    is_alphanumeric(b) || is_token_char(b) || b"~#@:".contains(&b)
}
```
and token is defined in RFC3261 as
```
token       =  1*(alphanum / "-" / "." / "!" / "%" / "*"
                     / "_" / "+" / "`" / "'" / "~" )
```

Looking better at it, makes more sense to add the new chars in `is_request_uri_char` and not in `is_token_char`.

Reply:
Thanks. These look rather uri than token indeed.
And token seems to be missing `~`